api/src/modules/maintenance-alerts/services/maintenance-alerts.service.ts

```diff
@@ -0,0 +1,199 @@
+import { IPagination, Pagination } from "@studiohyperdrive/pagination";
+import { Logger, NotFoundException } from "@nestjs/common";
+import { isArray } from "class-validator";
+import { isEmpty, isNil, set } from "lodash";
+
+import { DataService } from "../../data";
+import { CreateMaintenanceAlertDto, MaintenanceAlertsQueryDto, UpdateMaintenanceAlertDto } from './../dto/maintenance-alerts.dto';
+import { GqlMaintenanceAlert, MaintenanceAlert } from "../maintenance-alerts.types";
+import { ORDER_PROP_TO_DB_PROP } from "../maintenance-alerts.conts";
+
+import { SortDirection } from '../../shared/types';
+import { PaginationHelper } from "../../shared/helpers/pagination";
+import {
+	DeleteMaintenanceAlertDocument,
+	DeleteMaintenanceAlertMutation,
+	DeleteMaintenanceAlertMutationVariables,
+	FindMaintenanceAlertByIdDocument,
+	FindMaintenanceAlertByIdQuery,
+	FindMaintenanceAlertByIdQueryVariables,
+	FindMaintenanceAlertsDocument,
+	FindMaintenanceAlertsQuery,
+	FindMaintenanceAlertsQueryVariables,
+	InsertMaintenanceAlertDocument,
+	InsertMaintenanceAlertMutation,
+	InsertMaintenanceAlertMutationVariables,
+	UpdateMaintenanceAlertDocument,
+	UpdateMaintenanceAlertMutation,
+	UpdateMaintenanceAlertMutationVariables
+} from "../../shared/generated/graphql-db-types-hetarchief";
+
+export class MaintenanceAlertsService {
+	private logger: Logger = new Logger(MaintenanceAlertsService.name, { timestamp: true });
+
+	constructor(protected dataService: DataService) {}
+
+	public adapt(graphqlMaintenanceAlert: GqlMaintenanceAlert, isPersonal: boolean = false): MaintenanceAlert | null {
+		if (!graphqlMaintenanceAlert) {
+			return null;
+		}
+
+		let adaptedMaintenanceAlert: MaintenanceAlert = {
+			id: graphqlMaintenanceAlert.id,
+			title: graphqlMaintenanceAlert.title,
+			message: graphqlMaintenanceAlert.message,
+			type: graphqlMaintenanceAlert.type,
+			fromDate: graphqlMaintenanceAlert.from_date,
+			untilDate: graphqlMaintenanceAlert.until_date
+		}
+
+		// Brecht - userGroups should not be returned when /personal endpoint is called
+		if (!isPersonal) {
+			adaptedMaintenanceAlert = {
+				...adaptedMaintenanceAlert,
+				userGroups: graphqlMaintenanceAlert.user_groups
+			}
+		}
+
+		return adaptedMaintenanceAlert;
+	}
+
+	public async findAll(
+		inputQuery: MaintenanceAlertsQueryDto,
+		parameters: {
+			userGroupIds: (string | number)[],
+			isPersonal: boolean
+		} = null
+	): Promise<IPagination<MaintenanceAlert>> {
+		const { fromDate, untilDate, page, size, orderProp, orderDirection } = inputQuery;
+		const { offset, limit } = PaginationHelper.convertPagination(page, size);
+
+		/** Dynamically build the where object  */
+		const where: FindMaintenanceAlertsQueryVariables['where'] = {};
+
+		// Brecht - user group & special user group
+		if (!isNil(parameters)) {
```

Comment:
preference: flip the if else, so the condition is not a negative
```
if (!isNil())
   A
else
  B
```

to
```
if (isNil())
   B
else
  A
```

Non negative conditions are easier to reason about